File: app/src/main/java/posidon/launcher/feed/notifications/SwipeToDeleteCallback.java
package posidon.launcher.feed.notifications;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.ItemTouchHelper;
import androidx.recyclerview.widget.RecyclerView;

import org.jetbrains.annotations.NotNull;

public class SwipeToDeleteCallback extends ItemTouchHelper.SimpleCallback {
    public static SwipeListener swipeListener;

    public SwipeToDeleteCallback() { super(0,ItemTouchHelper.LEFT | ItemTouchHelper.RIGHT); }

    @Override
    public boolean onMove(@NonNull RecyclerView recyclerView, @NonNull RecyclerView.ViewHolder viewHolder, @NonNull RecyclerView.ViewHolder target) { return false; }

    @Override
    public void onSwiped(@NotNull RecyclerView.ViewHolder viewHolder, int direction) { swipeListener.onSwipe(viewHolder, direction); }

    public interface SwipeListener { void onSwipe(RecyclerView.ViewHolder viewHolder, int direction); }
}

File: app/src/main/java/posidon/launcher/feed/notifications/SwipeToDeleteCallback.kt
package posidon.launcher.feed.notifications

import androidx.recyclerview.widget.ItemTouchHelper
import androidx.recyclerview.widget.RecyclerView

class SwipeToDeleteCallback : ItemTouchHelper.SimpleCallback(0, ItemTouchHelper.LEFT or ItemTouchHelper.RIGHT) {
    override fun onMove(recyclerView: RecyclerView, viewHolder: RecyclerView.ViewHolder, target: RecyclerView.ViewHolder): Boolean {
        return false
    }

    override fun onSwiped(viewHolder: RecyclerView.ViewHolder, direction: Int) {
        swipeListener?.onSwipe(viewHolder, direction)
    }

    interface SwipeListener {
        fun onSwipe(viewHolder: RecyclerView.ViewHolder?, direction: Int)
    }

    companion object {
        var swipeListener: SwipeListener? = null
    }

    private var lastViewHolderToBeDragged: RecyclerView.ViewHolder? = null
    override fun onSelectedChanged(viewHolder: RecyclerView.ViewHolder?, actionState: Int) {
        super.onSelectedChanged(viewHolder, actionState)
        lastViewHolderToBeDragged?.itemView?.animate()?.alpha(1f)?.duration = 150L
        viewHolder?.itemView?.animate()?.alpha(0.9f)?.duration = 150L
        lastViewHolderToBeDragged = viewHolder
    }
}